zigpy/zcl/clusters/general.py

```diff
@@ -1274,10 +1277,160 @@ class PollControl(Cluster):
+        0x0007: ("translation_table_update_command", (t.Struct,), False),
+        0x0008: ("translation_table_request", (t.bitmap8,), False),
+        0x0009: ("pairing_configuration", (t.Struct,), False),
+    }
+    command={
+        0x00 : ("Identify",()),
+        0x10 : ("Scene 0",()),
+        0x11 : ("Scene 1",()),
+        0x12 : ("Scene 2",()),
+        0x13 : ("Scene 3",()),
+        0x14 : ("Scene 4",()),
+        0x15 : ("Scene 5",()),
+        0x17 : ("Scene 7",()),
+        0x16 : ("Scene 6",()),
+        0x18 : ("Scene 8",()),
+        0x19 : ("Scene 9",()),
+        0x1A : ("Scene 10",()),
+        0x1B : ("Scene 11",()),
+        0x1C : ("Scene 12",()),
+        0x1D : ("Scene 13",()),
+        0x1F : ("Scene 15",()),
+        0x1E : ("Scene 14",()),
+        0x20 : ("Off",()),
+        0x21 : ("On",()),
+        0x22 : ("Toggle",()),
+        0x23 : ("Release",()),
+        0x30 : ("Move Up",(t.Optional(t.uint8_t),)),
+        0x31 : ("Move Down",(t.Optional(t.uint8_t),)),
+        0x32 : ("Step Up",(t.uint8_t,t.Optional(t.uint16_t))),
+        0x33 : ("Step Down",(t.uint8_t,t.Optional(t.uint16_t))),
+        0x34 : ("Level Control",()),
+        0x35 : ("Move Up (with On/Off)",(t.Optional(t.uint8_t),)),
+        0x36 : ("Move Down (with On/Off)",(t.Optional(t.uint8_t),)),
+        0x37 : ("Step Up (with On/Off)",(t.uint8_t,t.Optional(t.uint16_t))),
+        0x38 : ("Step Down (with On/Off)",(t.uint8_t,t.Optional(t.uint16_t))),
+        0x40 : ("Move Hue Stop",()),
+        0x41 : ("Move Hue Up Color",(t.Optional(t.uint8_t),)),
+        0x42 : ("Move Hue Down Color",(t.Optional(t.uint8_t),)),
+        0x43 : ("Step Hue Up Color",(t.uint8_t,t.Optional(t.uint16_t))),
+        0x44 : ("Step Hue Down Color",(t.uint8_t,t.Optional(t.uint16_t))),
+        0x46 : ("Move Saturation Up",(t.Optional(t.uint8_t),)),
+        0x47 : ("Move Saturation Down",(t.Optional(t.uint8_t),)),
+        0x48 : ("Step Saturation Up",(t.Optional(t.uint8_t),)),
+        0x49 : ("Step Saturation Down",(t.Optional(t.uint8_t),)),
+        0x4A : ("Move Color",(t.uint16_t,t.uint16_t)),
+        0x4B : ("Step Color",(t.uint16_t,t.uint16_t,t.Optional(t.uint16_t))),
+        0x45 : ("Move Saturation Stop",()),
+        0x50 : ("Lock Door",()),
+        0x51 : ("Unlock Door",()),
+        0x60 : ("Press 1 of 1",()),
+        0x61 : ("Release 1 of 1",()),
+        0x62 : ("Press 1 of 2",()),
+        0x63 : ("Release 1 of 2",()),
+        0x64 : ("Press 2 of 2",()),
+        0x65 : ("Release 2 of 2",()),
+        0x66 : ("Short press 1 of 1",()),
+        0x67 : ("Short press 1 of 2",()),
+        0x68 : ("Short press 2 of 2",()),
+    }
+
+    zigBeeLinkKey=t.KeyData([0x5A, 0x69, 0x67, 0x42, 0x65, 0x65, 0x41, 0x6C, 0x6C, 0x69, 0x61, 0x6E, 0x63, 0x65, 0x30, 0x39])
+
+    def handle_message(self,data):
+        if data[0][0] == 12 and data[0][5] == 224 :
+            src_id = int.from_bytes(data[0][1:5],byteorder='little')
```

Comment:
is the data here a ZGP frame inside ZCL frame or vice versa? 
you also could do `data = data[0]`
and please do `pre-commit install` in you `zigpy` local folder, so it formats files with black

Reply:
Ok for `data = data[0]`

But I don't know what is `pre-commit install `

Reply:
1. activate your python venv if you are using any
2. `pip3 install -U pre-commit`
3. `cd <your local zigpy repo>`
4. `pre-commit install`

Reply:
Ok I need to test thaht (I dev on windows and auto upload file to my dev device)